### DISCUSSION TITLE
Update documentation on how to fetch info in zustand in React components on page load

### DISCUSSION THREAD

**Discussion by zsid** [pmndrs/zustand#1619] [General]
Hi. I am a new comer to the library and really struggling to find a best practice on fetching information on page lead.

The example for fetching is:
```
const useFishStore = create((set) => ({
  fishies: {},
  fetch: async (pond) => {
    const response = await fetch(pond)
    set({ fishies: await response.json() })
  },
}))
```
However, every single example/blog post that I have found (I have read more than 15), just fetch on button click. What i want to do is to fetch on page render. The obvious solution is to use `useEffect` and call that fetch in there. However, is this the way to do it? I really havent seen a single example with this and to me that is a common pattern.

**Comment by dai-shi** [comment]
useEffect is a valid solution. It's fired after React renders component and browser renders. 

It's not just about zustand but in general fetching data in useEffect is suboptimal. We could have started before React starting rendering. A solution can be onload or route change event, or even at module initialization. If you are unsure, useEffect is fine until the community suggests a solution.
